### ISSUE TITLE
lx syscall issue with multicast traffic and syscall IP_MULTICAST_TTL

### ISSUE THREAD

**Issue by failedrequest** [joyent/illumos-joyent#115]
In platform 20160512T071413Z when running a lx-zone , in my case it was both a home built CentOS 6.7 image as well as a official CentOS 6.7 (66208698-fb59-11e5-931f-e779314e6ea5) , Multicast traffic generated by omping is still failing to work correctly. This was supposedly fixed back in 2015 in os-4306 https://github.com/joyent/illumos-joyent/commit/70fe4e726e40ffd9c0eabfcc9a4477c54d181a63

Here is the strace output from omping
```


socket(PF_NETLINK, SOCK_RAW, 0)         = 3
bind(3, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 0
getsockname(3, {sa_family=AF_NETLINK, pid=5068, groups=00000000}, [12]) = 0
time(NULL)                              = 1463787505
sendto(3, "\24\0\0\0\26\0\1\3\361\237?W\0\0\0\0\0\0\0\0", 20, 0, {sa_family=AF_NETLINK, pid=0, groups=00000000}, 12) = 20
recvmsg(3, {msg_name(12)={sa_family=AF_NETLINK, pid=0, groups=00000000}, msg_iov(1)=[{" \0\0\0\24\0\2\0\361\237?W\314\23\0\0\2\10\0\376\1\0\0\0\10\0\1\0\177\0\0\1", 4096}], msg_controllen=0, msg_flags=0}, 0) = 32
recvmsg(3, {msg_name(12)={sa_family=AF_NETLINK, pid=0, groups=00000000}, msg_iov(1)=[{" \0\0\0\24\0\2\0\361\237?W\314\23\0\0\2\30\0\310\2\0\0\0\10\0\1\0\300\250\1\202", 4096}], msg_controllen=0, msg_flags=0}, 0) = 32
recvmsg(3, {msg_name(12)={sa_family=AF_NETLINK, pid=0, groups=00000000}, msg_iov(1)=[{",\0\0\0\24\0\2\0\361\237?W\314\23\0\0\n\200\0\376\1\0\0\0\24\0\1\0\0\0\0\0"..., 4096}], msg_controllen=0, msg_flags=0}, 0) = 44
recvmsg(3, {msg_name(12)={sa_family=AF_NETLINK, pid=0, groups=00000000}, msg_iov(1)=[{",\0\0\0\24\0\2\0\361\237?W\314\23\0\0\n\n\0\375\2\0\0\0\24\0\1\0\376\200\0\0"..., 4096}], msg_controllen=0, msg_flags=0}, 0) = 44
recvmsg(3, {msg_name(12)={sa_family=AF_NETLINK, pid=0, groups=00000000}, msg_iov(1)=[{"\20\0\0\0\3\0\2\0\361\237?W\314\23\0\0", 4096}], msg_controllen=0, msg_flags=0}, 0) = 16
close(3)                                = 0
socket(PF_INET, SOCK_DGRAM, IPPROTO_IP) = 3
getsockopt(3, SOL_SOCKET, SO_SNDBUF, [57344], [4]) = 0
getsockopt(3, SOL_SOCKET, SO_RCVBUF, [57344], [4]) = 0
setsockopt(3, SOL_IP, IP_TTL, [64], 4)  = 0
setsockopt(3, SOL_IP, IP_RECVTTL, [1], 4) = 0
setsockopt(3, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
setsockopt(3, SOL_IP, IP_MULTICAST_TTL, "@", 1) = -1 EINVAL (Invalid argument)
write(2, "omping: ", 8omping: )                 = 8
write(2, "Can't create/bind unicast socket", 32Can't create/bind unicast socket) = 32
write(2, ": ", 2: )                       = 2
write(2, "Invalid argument\n", 17Invalid argument
)      = 17
exit_group(1)                           = ?
+++ exited with 1 +++
```
Here is the output from the dtrace script noted in os-4306 which is printing 
lx-syscall::setsockopt:entry

```
  5111               omping     0 12
  5111               omping     1 29
  5111               omping     0 33
  5111               omping     0 34
  5111               omping     0 32
  5111               omping     0 33
  5111               omping     0 12
  5111               omping     1 29
  5111               omping     1 2
  5111               omping     0 34
  5111               omping     0 35
```

**Comment by pfmooney** [comment]
After taking a look at the setsockopt emulation, I believe I've found the issue.
If I cut a custom PI with the fix, would you want to test with it to ensure that it works?

**Comment by pfmooney** [comment]
Open as [OS-5428](https://smartos.org/bugview/OS-5428)

**Comment by failedrequest** [comment]
Yes , that's how I roll .

**Comment by pfmooney** [comment]
This should be fixed in 430df2d8b04c918ae55249a677210a90ba43b9d4
